### ISSUE TITLE
Feature request: merge multiple async iterators

### ISSUE THREAD

**Issue by pietercolpaert** [RubenVerborgh/AsyncIterator#11]
A function `merge` on an AsyncIterator would be a great tool to e.g., merge sort different iterators:
```javascript
require('async-iterator').merge([iterators], function (a, b) { return a.departureTime < b.departureTime})
```

Which element is emitted next is based on a boolean condition.

**Comment by RubenVerborgh** [comment]
I think @rubensworks made a library that does this. Might be able to steal that code.

**Comment by rubensworks** [comment]
Yes, I made [asynciterator-union](https://www.npmjs.com/package/asynciterator-union), which takes the union of two streams, but it doesn't do sorting.
It should not be too hard to extend it with a comparator-based iterator though.

(Note to self: we may need a new abstraction, where both `RoundRobinUnionIterator` and `MergeSortUnionIterator` extend from it)

**Comment by RubenVerborgh** [comment]
Now also available in this library itself: f405a7d201a1a82d5dd25bdb3c3209b89241f84f
